### PR TITLE
Update package with ROS transforms and refactor

### PR DESCRIPTION
**Pull request by dean4ta**
Class Diagram:
image

Driver state flow:
image

Rigid bodies can now be tracked with ROS transforms as specified in
config file, and these rigid bodies can be visualized in the same
perspective as viewed in Motive. Unnecessary dependencies removed and
the main node is no longer inheriting from
mocap_control::ControlledLifeCycleNode (which inherits from
rclcpp_lifecycle::LifecycleNode) and now inherits from LifecycleNode
which is a ROS2 standard feature.

- [x] Transform support for rigid bodies
- [x] better visualization in ROS
- [x] updating configuration file
- [x] removing unnecessary dependencies

The remaining objectives will be tackled in the next PR.
- [ ] get timing data from NatNet instead of from ROS
- [ ] clang formatting?
- [ ] squash commits before merging
- [ ] update documentation

### CODE REVIEW

File: mocap_optitrack_driver/src/mocap_optitrack_driver/mocap_optitrack_driver.cpp [hunk 1]
```diff
@@ -69,6 +65,69 @@ void OptitrackDriverNode::set_settings_optitrack()
...
+      id_rigid_body_map[rigid_body->ID] = rigid_body->szName;
+      rigid_body_id_map[rigid_body->szName] = rigid_body->ID;
+    }
+  }
+}
+
+void OptitrackDriverNode::get_rigid_bodies_from_params()
+{
+  const std::vector<std::string> prefix{"rigid_bodies"};
+  const std::string name_with_dot = std::string(this->get_name()) + ".";
```
Comment (popowskm) [inline mocap_optitrack_driver/src/mocap_optitrack_driver/mocap_optitrack_driver.cpp]: Is this variable unused?

File: mocap_optitrack_driver/include/mocap_optitrack_driver/mocap_optitrack_driver.hpp [hunk 2]
```diff
@@ -94,16 +105,28 @@ class OptitrackDriverNode : public mocap_control::ControlledLifecycleNode
   sFrameOfMocapData latest_data;
   sRigidBodyData latest_body_frame_data;
 
+  std::unordered_map<int, std::string> id_rigid_body_map;
+  std::unordered_map<std::string, int> rigid_body_id_map;
+  std::set<std::string> tf_frames_to_publish;
```
Comment (popowskm) [inline mocap_optitrack_driver/include/mocap_optitrack_driver/mocap_optitrack_driver.hpp]: Based on my understanding of the code, tf_frames_to_publish would be better named something like tf_rigid_bodies_to_publish.

File: mocap_optitrack_driver/config/mocap_optitrack_driver_params.yaml
Comment (popowskm) [inline mocap_optitrack_driver/config/mocap_optitrack_driver_params.yaml:1]: Is this file supposed to be included in the repo? Also, per a conversation I had with Morgan, Unicast is probably the preferred connection type when possible.

**Review by popowskm — COMMENTED** [review]
Aside from some super minor nitpicks, this looks pretty good to me. I don't have deep knowledge of ROS, but what I understood made sense to me.

File: mocap_optitrack_driver/src/mocap_optitrack_driver/mocap_optitrack_driver.cpp (hunk 1 above)
Comment (dean4ta) [reply mocap_optitrack_driver/src/mocap_optitrack_driver/mocap_optitrack_driver.cpp]: It is, removing

File: mocap_optitrack_driver/config/mocap_optitrack_driver_params.yaml
Comment (dean4ta) [reply mocap_optitrack_driver/config/mocap_optitrack_driver_params.yaml:1]: This file needs to be included in the repo. It is where the ros parameters are loaded from and acts as a template for other users. However, I do need to get rid of the changes that are machine specific and only include the changes that affect how the code runs

File: mocap_optitrack_driver/config/rigid_body_transform_view.rviz [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      Name: TF
+      Show Arrows: true
+      Show Axes: true
+      Show Names: true
+      Tree:
+        map:
+          optitrack:
+            Rigid Body 001:
+              {}
+            dean_rb:
```
Comment (popowskm) [inline mocap_optitrack_driver/config/rigid_body_transform_view.rviz]: There's a couple places in here that have "dean_rb". Without understanding much of how an rviz file works, that seems like something that should remain a local change.

**Review by popowskm — COMMENTED** [review]
Looks pretty good to me. Only one minor matter of style.

File: mocap_optitrack_driver/config/rigid_body_transform_view.rviz (hunk 3 above)
Comment (popowskm) [reply mocap_optitrack_driver/config/rigid_body_transform_view.rviz]: It looks like it's referring to the rigid body in the YAML file. Is there a good way to pull that stuff out, or could we at least rename it more generically?

Comment (dean4ta) [reply mocap_optitrack_driver/config/rigid_body_transform_view.rviz]: I removed those variables from the rviz file and from the yaml file. The rigid bodies are now Rigid Body 001 and 002

**Review by popowskm — APPROVED** [review]

**Pull request merged**